src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs

```diff
@@ -106,16 +106,18 @@ public OptimismEthRpcModule(
         return GetBlockReceipts(_receiptFinder, blockParameter, _blockFinder, _specProvider, _opSpecHelper);
     }
 
-    public override async Task<ResultWrapper<Hash256>> eth_sendTransaction(TransactionForRpc rpcTx)
+    public async Task<ResultWrapper<Hash256>> eth_sendTransaction(OptimismTransactionForRpc rpcTx)
     {
-        Transaction tx = rpcTx.ToTransactionWithDefaults(_blockchainBridge.GetChainId());
+        Transaction tx = rpcTx.ToTransaction();
+        tx.ChainId = _blockchainBridge.GetChainId();
         tx.SenderAddress ??= _ecdsa.RecoverAddress(tx);
 
         if (tx.SenderAddress is null)
         {
             return ResultWrapper<Hash256>.Fail("Failed to recover sender");
         }
 
+        // TODO: Optimism transactions do not have a Nonce (it's always 0 when queried)
```

Comment:
Not sure what's the intention behind the orignal code. In the original `op-geth` codebase `Nonce` is implemented as method (not a field) that always returns `0` (see https://github.com/ethereum-optimism/op-geth/blob/8af19cf20261c0b62f98cc27da3a268f542822ee/core/types/deposit_tx.go#L79). Why are we setting a particular nonce here?